pkg/apis/core/validation/validation.go

```diff
@@ -4688,6 +4688,11 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 	allErrs = append(allErrs, validatePodMetadataAndSpec(newPod, opts)...)
 	allErrs = append(allErrs, ValidatePodSpecificAnnotationUpdates(newPod, oldPod, fldPath.Child("annotations"), opts)...)
 
+	// static pods don't support ephemeral containers #113935
+	if _, ok := oldPod.Annotations[core.MirrorPodAnnotationKey]; ok {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("annotations"), fmt.Sprintf("static pods (%q) do not support ephemeral containers", oldPod.ObjectMeta.GetName())))
```

Comment:
the annotation check lgtm, and ValidatePodEphemeralContainersUpdate is *only* called when adding a new ephemeral container, so making this function stricter will not break any existing mirror pods that happened to have ephemeral containers added previously

If we do encounter a mirror pod, this error is not one of many field errors that can be "corrected", it means this pod is not eligible to have ephemeral containers added to it.

For that reason, I would probably return a solo forbidden error here instead of appending to `allErrs`, and would set the field path to `field.NewPath("")` rather than pointing at the annotation specifically